cluster/gce/configure-vm.sh

```diff
@@ -260,6 +260,13 @@ function create-salt-auth() {
   kubelet_auth_file="/srv/salt-overlay/salt/kubelet/kubernetes_auth"
   (umask 077;
     echo "{\"BearerToken\": \"${KUBELET_TOKEN}\", \"Insecure\": true }" > "${kubelet_auth_file}")
+
+  # Generate tokens for other "service accounts".  Append to known_tokens.
+  for a in scheduler controller_manager logging monitoring dns
+  do
+    token=$(dd if=/dev/urandom bs=128 count=1 2>/dev/null | base64 | tr -d "=+/" | dd bs=32 count=1 2>/dev/null)
+    echo "${token},kubernetes_builtin:$a,kubernetes_builtin:$a" >> "${known_tokens_file}"
```

Comment:
We had very specific users names and we are using the system: prefix to scope deeper roles.  

https://github.com/openshift/origin/blob/master/pkg/cmd/server/bootstrappolicy/constants.go
https://github.com/openshift/origin/blob/master/pkg/cmd/server/admin/default_certs.go

Would be good to settle on a pattern for prefixing as we go (don't have to change here).

> On Mar 27, 2015, at 7:18 PM, Jordan Liggitt <notifications@github.com> wrote:
> 
> In cluster/gce/configure-vm.sh:
> 
> > @@ -260,6 +260,13 @@ function create-salt-auth() {
> >    kubelet_auth_file="/srv/salt-overlay/salt/kubelet/kubernetes_auth"
> >    (umask 077;
> >      echo "{\"BearerToken\": \"${KUBELET_TOKEN}\", \"Insecure\": true }" > "${kubelet_auth_file}")
> > +
> > +  # Generate tokens for other "service accounts".  Append to known_tokens.
> > +  for a in scheduler controller_manager logging monitoring dns
> > +  do
> > +    token=$(dd if=/dev/urandom bs=128 count=1 2>/dev/null | base64 | tr -d "=+/" | dd bs=32 count=1 2>/dev/null)
> > +    echo "${token},kubernetes_builtin:$a,kubernetes_builtin:$a" >> "${known_tokens_file}"
> We ended up with normal usernames being unnamedspaced (so they can inherit the preferred username from the bedrock identity provider), and namespaced system/service account usernames with a "prefix:" (like "system:" or "project:"). That means ":" is disallowed in normal usernames for us.
> 
> —
> Reply to this email directly or view it on GitHub.
>